library/src/main/java/com/pokegoapi/auth/PtcCredentialProvider.java

```diff
@@ -147,11 +148,15 @@ public PtcCredentialProvider(OkHttpClient client, String username, String passwo
 	 */
 	private void login(String username, String password, int attempt)
 			throws LoginFailedException, InvalidCredentialsException {
-
 		try {
+			HttpUrl localObject = HttpUrl.parse("https://sso.pokemon.com/sso/oauth2.0/authorize").newBuilder()
```

Comment:
Since the entire statement falls under the same try block and if this part fails, so does the entire thing, you don't even need a variable.
`url(HttpUrl.parse("https://sso.pokemon.com/sso/oauth2.0/authorize").newBuilder().addQueryParameter("client_id",  CLIENT_ID).addQueryParameter("redirect_uri", REDIRECT_URI).addQueryParameter("locale", "en").build())`